### ISSUE TITLE
Consider how to best manage our SQL statements

### ISSUE THREAD

**Issue by linacambridge** [mozilla/application-services#570]
I just learned about `include_str!` from the [Diesel docs](http://diesel.rs/):

```rust
// Using `include_str!` allows us to keep the SQL in a
// separate file, where our editor can give us SQL specific
// syntax highlighting.
sql_query(include_str!("complex_users_by_organization.sql"))
    .bind::<Integer, _>(organization_id)
    .bind::<BigInt, _>(offset)
    .bind::<BigInt, _>(limit)
    .load::<User>(conn)?;
```

This might be useful for some of our larger table schema definitions, and the bookmark merging triggers.

**Comment by thomcc** [comment]
I'm positive about this, but the issue here is mostly around that we like to use format! to customize them in some places.

**Comment by mhammond** [comment]
While `include_str!` might not be the answer, I think we should find something better than what we have. We have schema.rs which uses mozilla-central style "const SOME_SHOUTY_NAME" while much of the rest of the code just uses SQL directly inline. In an ideal world, we'd have something that (a) makes it easy to find what SQL is used where and (b) enables the SQL to be used in other contexts (eg, via "EXPLAIN QUERY PLAN") without copy/pasting stuff somewhere.

I've no idea what the answer to that is, so my contribution is to update the title of this issue :) Thanks for raising it Lina!

**Comment by linacambridge** [comment]
Fixed in #671!
